pkg/operator/tektonchains.go

```diff
@@ -45,4 +57,137 @@ func EnsureTektonChainsExists(clients chainv1alpha.TektonChainInterface, names u
+	fmt.Println("Waiting 2 minutes for images to appear in image registry")
+	cmd.MustSuccedIncreasedTimeout(time.Second*130, "sleep", "120")
+}
+
+func GetImageDigestedUrl() (string, string) {
+	// Get Image digest
+	var imageDigest string
+	jsonOutput := cmd.MustSucceed("tkn", "tr", "describe", "--last", "-o", "json").Stdout()
+	// Parse Json Output
+	type TaskRun struct {
+		Status struct {
+			Results []struct {
+				Name  string `json:"name"`
+				Value string `json:"value"`
+			} `json:"results"`
+		} `json:"status"`
+	}
+	var taskrun TaskRun
+	err := json.Unmarshal([]byte(jsonOutput), &taskrun)
+	if err != nil {
+		testsuit.T.Errorf("Error parsing Json output")
+	}
+
+	// Get IMAGE_DIGEST value
+	for _, result := range taskrun.Status.Results {
+		if strings.Contains(result.Name, "IMAGE_DIGEST") {
+			imageDigest = strings.Split(result.Value, ":")[1]
+		}
+	}
+
+	// Return image url with digest
+	url := fmt.Sprintf("%s@sha256:%s", repo, imageDigest)
+	return url, imageDigest
+}
+
+func VerifyImageSignature() {
+	url, _ := GetImageDigestedUrl()
+	cmd.MustSucceed("cosign", "verify", "--key", public_key_path+"/cosign.pub", url)
+}
+
+func VerifyAttestation() {
+	url, _ := GetImageDigestedUrl()
+	cmd.MustSucceed("cosign", "verify-attestation", "--key", public_key_path+"/cosign.pub", "--type", "slsaprovenance", url)
+}
+
+func CheckAttestation() {
+	// Get UUID
+	_, imageDigest := GetImageDigestedUrl()
+	jsonOutput := cmd.MustSucceed("rekor-cli", "search", "--format", "json", "--sha", imageDigest).Stdout()
+
+	// Parse Json output to find UUID
+	type UUID struct {
+		UUIDs []string `json:"UUIDs"`
+	}
+	var uuid UUID
+	err := json.Unmarshal([]byte(jsonOutput), &uuid)
+	if err != nil {
+		testsuit.T.Errorf("Error parsing Json output")
+	}
+	rekor_uuid := uuid.UUIDs[0]
+
+	// Check the Attestation
+	if strings.Contains(cmd.Run("rekor-cli", "get", "--uuid", rekor_uuid).Stdout(), "getLogEntryByUuidNotFound") {
+		testsuit.T.Errorf("Failed to find Attestation")
+	}
+}
+
+func CreateFileWithCosignPubKey() {
+	chainsPublicKey := cmd.MustSucceed("oc", "get", "secrets", "signing-secrets", "-n", "openshift-pipelines", "-o", "jsonpath='{.data.cosign\\.pub}'").Stdout()
+	chainsPublicKey = strings.Trim(chainsPublicKey, "'")
+	decodedPublicKey, err := base64.StdEncoding.DecodeString(chainsPublicKey)
+	if err != nil {
+		testsuit.T.Errorf("Error decoding base64")
+	}
+	filepath := filepath.Join(public_key_path, "cosign.pub")
+	file, err := os.Create(filepath)
+	if err != nil {
+		testsuit.T.Errorf("Error creating file")
+	}
+	defer file.Close()
+	_, err = file.WriteString(string(decodedPublicKey))
+	if err != nil {
+		testsuit.T.Errorf("Error writing to file")
+	}
+}
+
+func CreateSigningSecretForTektonChains() {
+	chainsPublicKey := os.Getenv("CHAINS_COSIGN_PUBLIC")
+	chainsPrivateKey := os.Getenv("CHAINS_COSIGN_PRIVATE")
+	os.Setenv("COSIGN_PASSWORD", "chainstest")
+	chainsPassword := os.Getenv("COSIGN_PASSWORD")
```

Comment:
These two lines don't make sense bacause you are overwriting whatever the user provided. `os.Getenv("COSIGN_PASSWORD")` should go into first branch of `if` statement. `os.Setenv()` should go to the second branch of `if` statement.